### PR TITLE
Update EventGrid SDK t1 to t2

### PR DESCRIPTION
**Pull request by v-yilinhu**
1. Update EventGrid SDK t1 to t2

2. Update files:

- i ./AKVSQLRotation.cs
- ii ./KeyVault-Rotation-SQLPassword-Csharp.csproj

@jongio,@scottaddie,@AlexGhiondea,@jameszliao-msft,@ellismg and @ahamad-MS for notification.

### CODE REVIEW

**Comment by v-xuto** [comment]
@jongio,@scottaddie,@AlexGhiondea,@jameszliao-msft,@ellismg and @ahamad-MS  Could you please review this PR?

**Comment by v-xuto** [comment]
@jongio, @scottaddie, @AlexGhiondea, @jameszliao-msft, @ellismg, @ahamad-MS Could you please review this PR?

**Comment by v-xuto** [comment]
@scottaddie, @AlexGhiondea, @jameszliao-msft, @ellismg, @ahamad-MS Could you please review this PR?

**Comment by v-xuto** [comment]
@scottaddie, @AlexGhiondea, @jameszliao-msft, @ellismg, @ahamad-MS Could you please review this PR?

File: KeyVault-Rotation-SQLPassword-Csharp.csproj [hunk 1]
```diff
@@ -1,16 +1,17 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>netcoreapp3.1</TargetFramework>
     <AzureFunctionsVersion>v3</AzureFunctionsVersion>
     <RootNamespace>Microsoft.KeyVault</RootNamespace>
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="Azure.Messaging.EventGrid" Version="4.5.0" />
     <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EventGrid" Version="2.1.0" />
```
Comment (JoshLove-msft) [inline KeyVault-Rotation-SQLPassword-Csharp.csproj]:
```suggestion
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EventGrid" Version="3.0.1" />
```

File: KeyVault-Rotation-SQLPassword-Csharp.csproj [hunk 2]
```diff
@@ -1,16 +1,17 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>netcoreapp3.1</TargetFramework>
     <AzureFunctionsVersion>v3</AzureFunctionsVersion>
     <RootNamespace>Microsoft.KeyVault</RootNamespace>
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="Azure.Messaging.EventGrid" Version="4.5.0" />
```
Comment (JoshLove-msft) [inline KeyVault-Rotation-SQLPassword-Csharp.csproj]: This shouldn't be necessary, as it will be pulled in via the extension.

**Comment by v-xuto** [comment]
@JoshLove-msft  We have updated by your comments, please re-review this PR.

**Comment by v-xuto** [comment]
@JoshLove-msft We have updated by your comments, please re-review this PR.

File: KeyVault-Rotation-SQLPassword-Csharp.csproj [hunk 3]
```diff
@@ -1,16 +1,16 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>netcoreapp3.1</TargetFramework>
     <AzureFunctionsVersion>v3</AzureFunctionsVersion>
     <RootNamespace>Microsoft.KeyVault</RootNamespace>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EventGrid" Version="2.1.0" />
+    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EventGrid" Version="3.0.1" />
```
Comment (JoshLove-msft) [inline KeyVault-Rotation-SQLPassword-Csharp.csproj]:
```suggestion
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EventGrid" Version="3.1.0" />
```

Comment (v-yilinhu) [reply KeyVault-Rotation-SQLPassword-Csharp.csproj]: @JoshLove-msft I have updated, please review.

**Review by JoshLove-msft — APPROVED** [review]

**Comment by v-xuto** [comment]
@jongio Please help merge this PR.

**Pull request merged**